internal/access/credential.go

```diff
@@ -225,3 +230,30 @@ func checkPasswordRequirements(db data.ReadTxn, password string) error {
 
 	return nil
 }
+
+// checkBadPasswords checks if the password is a known bad password, i.e. a widely reused password.
+func checkBadPasswords(password string) error {
+	badPasswordsFile := os.Getenv("INFRA_SERVER_BAD_PASSWORDS_FILE")
+	if badPasswordsFile == "" {
+		return nil
+	}
+
+	file, err := os.OpenFile(badPasswordsFile, os.O_RDONLY|os.O_CREATE, 0o600)
+	if err != nil {
+		return err
+	}
+
+	scan := bufio.NewScanner(file)
+	scan.Split(bufio.ScanLines)
+	for scan.Scan() {
+		if scan.Text() == password {
+			return fmt.Errorf("%w: cannot use a password commonly found in data breaches", internal.ErrBadRequest)
```

Comment:
```suggestion
			return fmt.Errorf("%w: cannot use a password using common words", internal.ErrBadRequest)
```

Nit suggestion on text to be a tad friendlier